internal/release/main.go

```diff
@@ -29,29 +30,55 @@ func getProjectPath() (string, error) {
 	return filepath.Dir(filepath.Dir(wd)), nil
 }
 
-func validateModFile(path string) error {
+func validateModFile(path string, version *semver.Version) error {
 	// Ensure the path is consistent with the go mod
 	baseProjectPath, err := getProjectPath()
 	if err != nil {
 		return err
 	}
+
 	modPath := filepath.Join(baseProjectPath, path, "go.mod")
 	bytes, err := os.ReadFile(modPath)
 	if err != nil {
 		return err
 	}
+
 	modFile, err := modfile.Parse(modPath, bytes, nil)
 	if err != nil {
 		return err
 	}
+
+	actualModPath := modFile.Module.Mod.Path
+	var actualMajorVersion int64
+
+	// Strip the /vX from the end of any mod paths
+	actualModPathSections := strings.Split(actualModPath, "/")
+	if len(actualModPathSections) == 5 {
+		actualModPath = strings.Join(actualModPathSections[:4], "/")
+		actualMajorVersion, err = strconv.ParseInt(strings.TrimPrefix(actualModPathSections[4], "v"), 10, 0)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
this does assume that we will never do non top-level modules,

ie: `github.com/defenseunicorns/pkg/auth/retry` for example

Reply:
correct